neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/BoltNeo4jDialect.java

```diff
@@ -589,6 +592,36 @@ public void forEachTuple(ModelConsumer consumer, TupleTypeContext tupleTypeConte
 		consumer.consume( tupleSupplier );
 	}
 
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(
+			String storedProcedureName, ProcedureQueryParameters queryParameters, TupleContext tupleContext) {
+		Map.Entry<String, Map> queryAndParams = buildProcedureQueryWithParams( storedProcedureName, queryParameters );
+		Transaction transaction = transaction( tupleContext );
+		StatementResult result;
+		if ( transaction == null ) {
+			DatastoreProvider datastoreProvider = getServiceRegistry().getService( DatastoreProvider.class );
+			BoltNeo4jDatastoreProvider neo4jProvider = (BoltNeo4jDatastoreProvider) datastoreProvider;
+			BoltNeo4jClient client = neo4jProvider.getClient();
+			try ( Session session = client.getDriver().session() ) {
+				result = session.run( queryAndParams.getKey(), queryAndParams.getValue() );
+			}
+		}
+		else {
+			result = transaction.run( queryAndParams.getKey(), queryAndParams.getValue() );
+		}
+		return CollectionHelper.newClosableIterator( extractTuples( result ) );
+	}
+
+	private List<Tuple> extractTuples(StatementResult result) {
+		List<Tuple> tuples = new ArrayList<>();
+		while ( result.hasNext() ) {
+			Tuple tuple = new Tuple();
+			result.next().asMap().forEach( (key, value) -> tuple.put( key, value ) );
+			tuples.add( tuple );
+		}
+		return tuples;
+	}
```

Comment:
I don't think you need this method, can't you use the following?
```
return new BoltNeo4jMapsTupleIterator( statementResult );
```
Check what we do for `executeBackendQuery`.